### ISSUE TITLE
Opening an editor window for replying to a message in Evolution mail reboots my machine

### ISSUE THREAD

**Issue by alexeymuranov** [NixOS/nixpkgs#133414]
### Describe the bug

I've received an email message from a friend today. I tried to reply to it by selecting a relevant part of the message and clicking "Group Reply" in [Evolution](https://wiki.gnome.org/Apps/Evolution), and my machine rebooted. I tried 2 more times, and it rebooted both times.

I tried clicking "Group Reply" on a different message, and it worked fine. I tried doing it on the "problematic" message without selecting any text, and it worked fine too.

Then I managed somehow to reply to the "problematic" message after selecting the relevant text. After that i experimented a bit more, and sometimes the reply editor window would open, but I got my machine rebooted 2 more times by performing the same action.

I am on NixOS 20.05 with Linux 5.10.57-hardened1.

Any help with debugging this behaviour will be appreciated.

### Steps To Reproduce
1. I launch Evolution,
2. I select a certain text in a certain message,
3. I click "Group Reply".

### Expected behavior
Should not reboot.

### Metadata
```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.57-hardened1, NixOS, 21.05.2362.b09c989b82f (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(alexey): `""`
 - channels(root): `"nixos-21.05.2362.b09c989b82f"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by 7c6f434c** [comment]
I have no idea at all what's going on, but maybe to start collecting data…

Have you looked in the system logs to check if anything possibly relevant is there right before the next boot?

Do you use Wayland or Xorg? If Wayland, which composer?

Can you forward the email to yourself? Does selecting the same part in the forwarded email and trying to reply also lead to reboot? (I wonder if selecting first half/second half and bisecting like that could lead to a somewhat understandable «selecting this specific text and replying crashes half of the time»… but no idea if specific text selected or message identity is the relevant part).

**Comment by alexeymuranov** [comment]
@7c6f434c, this is on Xorg.

When i have some time for this, i'll try to investigate if text selection is relevant and if i can reproduce this with a different message.

I am not experienced with system debugging, could you indicate how to check the relevant part of the system logs, please?

**Comment by 7c6f434c** [comment]
I think you could use something like `journalctl --since "2021-08-10 20:00:00" --until "2021-08-10 20:03:00"` (I don't use `journald` on my system, so I only remember some random subset of selection options)

**Comment by alexeymuranov** [comment]
I cannot reproduce the issue on Wayland.

[Here](https://gist.github.com/alexeymuranov/6851665ae170aebdcff9b3f758ddb2a6) is the output of `journalctl` for the 3s before the last hard reboot.

**Comment by alexeymuranov** [comment]
I managed to reproduce hard reboot without selecting any text, and also when trying to reply to a different message.

**Comment by 7c6f434c** [comment]
The logs show a shutdown of the session (is UID 132 your user account?) with something left of Evolution still trying to reach something till the reboot. Does it correspond to anything visible on the screen? How long between trying to reply and the reboot?

**Comment by alexeymuranov** [comment]
No, my UID is 1000. The reboot is immediate (hard shutdown), so it looks like there is nothing directly relevant to the reboot in the log.

**Comment by 7c6f434c** [comment]
Hmmm, what GPU driver in use?

**Comment by alexeymuranov** [comment]
```nix
# /etc/nixos/configuration.nix
{ config, pkgs, ... }:

{ # ...
  services = { # ...
    xserver = { # ...
      videoDrivers = [ "intel" "modesetting" ];
    };
  };
}
```

**Comment by 7c6f434c** [comment]
Are you sure you want «intel» here? https://github.com/NixOS/nixpkgs/pull/55583

**Comment by alexeymuranov** [comment]
@7c6f434c, thanks for pointing this out to me. Would you mind indicating what would be the right setup for Intel GPU (Skylake HD Graphics 520)?

**Comment by 7c6f434c** [comment]
Just `modesetting` without `intel`, as mentioned in the release notes in the linked PR.

**Comment by alexeymuranov** [comment]
After removing `"intel"` from `services.xserver.videoDrivers`, and also `pkgs.vaapiIntel` and `pkgs.intel-ocl` from `hardware.opengl.extraPackages`, i could not reproduce the issue.

**Comment by alexeymuranov** [comment]
I tried to put back `pkgs.vaapiIntel` and `pkgs.intel-ocl`, and still could not reproduce the issue. So, probably, the cause was just in the Intel driver.

I am not sure I need OpenCL, but i would like to have it just in case.

Would you recommend any benchmarks to test different configurations? I ran `glmark2`, and it semed like removing `pkgs.vaapiIntel` and `pkgs.intel-ocl` slightly improved the score (with `modesetting` driver without `intel`).

**Comment by 7c6f434c** [comment]
I have no idea what are good benchmarks, no. Won't be surprised if asking an incompatible VAAPI driver if it can work is slightly worse than just not having it available. I would expect that the packages for OpenCL/VAAPI just won't do anything useful now; a replacement might work or not, no idea. Half of Nixpkgs is a case study in wanting some functionality available without having any clear testcase/regular usecase… (spoiler: it turns out it does not work as a result)
